### ISSUE TITLE
Can't paste when switching back keyboard

### ISSUE THREAD

**Issue by KLaurence** [florisboard/florisboard#2024]
#### Short description
When a user copies text while using another keyboard, the Paste button is unclickable and the user therefore can't paste text. This occurs even when the clipboard has text.

#### Steps to reproduce
1. Switch to Florisboard as the primary keyboard.
2. Go to a text box and copy some text. The Paste button should become clickable.
3. Clear the clipboard using the _Clear primary clip_ button.
4. Switch to another keyboard and copy some text.
5. Switch back to Florisboard. The Paste button should be not clickable.

#### Environment information
- FlorisBoard Version: [0.4.0-debug-c3fae09](https://github.com/florisboard/florisboard/commit/c3fae0971c69c6f2ebb88ddaad6f1185236c6627) (88)
- Install Source: GitHub
- Android: 10

**Comment by Glitchy-Tozier** [comment]
Are you suggesting that the content copied by another keyboard should be available in Florisboard? If yes:

1. I don't think this is possible if the other keyboard doesn't want to share that information
2. _However_, there is a setting that makes Florisboard read content copied via Android's copy-menu.
![Screenshot_20220826-152923_FlorisBoard Beta.jpg](https://user-images.githubusercontent.com/59611881/186914908-9090ae08-f194-4138-8be7-5063d8e71b9c.jpg)
In the settings, check this option:
![Screenshot_20220826-152530_FlorisBoard Beta.jpg](https://user-images.githubusercontent.com/59611881/186914872-8182733e-12af-49a1-9f01-b3600327b44e.jpg)
![Screenshot_20220826-152552_FlorisBoard Beta.jpg](https://user-images.githubusercontent.com/59611881/186914926-a7dc219f-d463-49e3-ac2e-484cacbb21c1.jpg)

**Comment by KLaurence** [comment]
I meant to mention that the text should be copied while using another keyboard but not necessarily via inbuilt functionality of that keyboard (the bug still happens when copying via the system Copy button that appears when selecting).

I'll attempt to record a video that illustrates the issue.

**Comment by Glitchy-Tozier** [comment]
Just to make sure: Did you try it after activating the setting I showed (the first option inside the clipboard-settings)?

**Comment by KLaurence** [comment]
Yep. Enabling the internal clipboard (and the options _Sync from system clipboard_ and _Sync to system clipboard_) doesn't solve it.

**Comment by patrickgold** [comment]
The reason is very simple why this does not work: FlorisBoard has only access to the clipboard as long as it is the default IME, else it gets "null" items delivered, which are useless. This is to protect the privacy of the clipboard data and is enforced by the Android system.

As FlorisBoard relies solely on the clipboard primary clip changed listener it thus does not notice a change if you use another keyboard. What I could try is every time the IME window is shown again to manually query the current system primary clip and compare it with the cache. If it matches I assume nothing changed, if it doesn't match I will treat this as a change event. Of course reading from the system clipboard is only done either if you have disabled the internal clipboard or if you have enabled it AND have enabled sync from system.

The corresponding update code can be seen here: https://github.com/florisboard/florisboard/blob/de2b3b9433ba56bc629d53e9538cea5446fea6ff/app/src/main/kotlin/dev/patrickgold/florisboard/ime/clipboard/ClipboardManager.kt#L162-L202

**Comment by KLaurence** [comment]
I would be okay with that solution. I also don't think there's a problem with assuming that no clipboard operations occurred if the stored clipboard is the same, since the Paste button would still be clickable (or if the clipboard is empty this entire time, unclickable) regardless.
